### PR TITLE
Added Next permutation Docs

### PR DESCRIPTION
**Pull request by NEERAJAP2001**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

YES

### Description
Added Next permutation Docs

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

Solves #26

### CODE REVIEW

**Comment by NEERAJAP2001** [comment]
Please review @HarshCasper

**Comment by NEERAJAP2001** [comment]
Any Updates @HarshCasper @iamrajiv ?

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@NEERAJAP2001 Fix typos like `Implemention`, `postion`, `Finnaly`.

**Review by iamrajiv — APPROVED** [review]

File: docs/mathematics/next-permutation.mdx [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+if __name__ == "__main__":
+    #User input for both the strings
+    print('Enter values of the array')
+    array = list(map(int, input().rstrip().split()))
+    print('Next Permutation is: ',nextPermutation(array))  
+
+```
+</TabItem>
+
+</Tabs>
```
Comment (HarshCasper) [inline docs/mathematics/next-permutation.mdx:105]: The Code does not even render on the Website. Did you do a proper check on your local machine before pushing the PR?

File: docs/mathematics/next-permutation.mdx [hunk 2]
```diff
@@ -0,0 +1,128 @@
+---
+title: Next Permutation
+slug: /mathematics/next-permutation
+---
+
+## Introduction
+
+Implementation of next permutation, it rearranges numbers into the lexicographically next greater permutation of numbers.
+If such an arrangement is not possible, it must rearrange it as the lowest possible order (sorted in ascending order).
```
Comment (HarshCasper) [inline docs/mathematics/next-permutation.mdx:9]: Wrap in a single line. Add more description.

File: docs/mathematics/next-permutation.mdx [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+---
+
+## Introduction
+
+Implementation of next permutation, it rearranges numbers into the lexicographically next greater permutation of numbers.
+If such an arrangement is not possible, it must rearrange it as the lowest possible order (sorted in ascending order).
+
+## Explanation 
+We need to find the next lexicographic permutation of the given list of numbers than the number formed by the given array.
+Our approach would be inplace.
```
Comment (HarshCasper) [inline docs/mathematics/next-permutation.mdx:13]: Wrap in a single line. Add more description.

File: docs/mathematics/next-permutation.mdx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+
+* Now, Over that process once the `pointer` is equal to `-1` then it simply means we have to return `array.reverse()` 
+
+* If it is not yet equal to `-1' then we iterate over the range of `length - 1, pointer` with incrementer as `-1` and variable as `number`.
+
+* Inside the loop we check condition if `array[pointer] < array[number]` is `True` then swap `array[pointer]` with `array[number]` and `break` after that is done.
+
+* Finally we index over our array from `pointer + 1` till `end` of array and simply reverse is using `reversed` function.
+
+* Now, we'd simply return the modified array as our answer.
```
Comment (HarshCasper) [inline docs/mathematics/next-permutation.mdx:51]: Remove blank lines

**Review by HarshCasper — CHANGES_REQUESTED** [review]

File: docs/mathematics/next-permutation.mdx (hunk 1 above)
Comment (NEERAJAP2001) [reply docs/mathematics/next-permutation.mdx:94]: But it works fine in the deployed version @HarshCasper. Can you please confirm?


Screenshot (286)

**Comment by NEERAJAP2001** [comment]
@HarshCasper  any updates on this PR?

**Review by ankitaggarwal23 — APPROVED** [review]

**Pull request merged**